r/R/duckdb.R

```diff
@@ -111,3 +111,42 @@ duckdb_disconnector <- function(con, tbl_name) {
   })
   environment()
 }
+
+to_arrow <- function(.data) {
+  res <- DBI::dbSendQuery(dbplyr::remote_con(.data), dbplyr::remote_query(.data), arrow = TRUE)
+
+  # transform the source ExecNode we get below into something dataset-like that we can
+  # make arrow_dplyr_querys/datasets from
+  # Should be able to:
+  # * get the record batch reader from duckdb
+  # * produce the SourceNode
+  # * build an ExecPlan with that in place of the ScanNode you would have gotten from ExecNode_Scan
+  # source_node <- ExecNode_ReadFromRecordBatchReader(plan, RBR)
+  .data <- duckdb::duckdb_fetch_record_batch(res)
+  structure(
```

Comment:
You shouldn't have to copy this from `arrow_dplyr_query()` should you?

Reply:
Right, I did out of expedience, but will come back and clean this up to do the right thing later.

Reply:
You might need to `collapse()` this further, judging from how you're handling this case in `plan$Scan`: you don't want any filter/projection at this level of nesting